gravitee-apim-console-webui/src/management/api/general-info/api-general-info.component.ts

```diff
@@ -111,7 +117,7 @@ export class ApiGeneralInfoComponent implements OnInit, OnDestroy {
 
     combineLatest([this.apiService.get(this.apiId), this.categoryService.list()])
       .pipe(
-        switchMap(([api, categories]) =>
+        switchMap(([api, categories]: [Api, Category[]]) =>
```

Comment:
Normally, the inferred type should be enough, no? if it's not good, there's probably a problem at the source of the type. 🤔

Reply:
yeah, it inferred the type correctly. The advantage for explicit typing type is easier navigation through the code as the type annotations work like hyperlinks, also we are more explicit what we are expecting.

Reply:
The problem is that sometimes this forces the type, which means that if the upstream type changes, the typing doesn't inform us that we've changed the contract. 
by example here if you add `[ApiV2, Category[]]` no error and yet it's not an ApiV2. 

It's up to your IDE to provide you with the means to navigate inferred types, but not to be added to the code IMHO